ghost/admin/app/services/slug-generator.js

```diff
@@ -10,15 +10,20 @@ export default class SlugGeneratorService extends Service {
     @service ghostPaths;
     @service ajax;
 
-    generateSlug(slugType, textToSlugify) {
+    generateSlug(slugType, textToSlugify, modelId) {
         let url;
 
         if (!textToSlugify) {
             return resolve('');
         }
 
         // We already do a partial slugify at the client side to prevent issues with Pro returning a 404 page because of invalid (encoded) characters (a newline, %0A, for example)
-        url = this.get('ghostPaths.url').api('slugs', slugType, encodeURIComponent(slugify(textToSlugify)));
+        let name = encodeURIComponent(slugify(textToSlugify));
+        if (modelId) {
+            url = this.get('ghostPaths.url').api('slugs', slugType, name, modelId);
+        } else {
+            url = this.get('ghostPaths.url').api('slugs', slugType, name);
```

Comment:
Are their reproducible scenarios where `modelId` may not exist given the changes?

Reply:
Yeah, this service is used for generating slugs for tags and users too!